File: 66-plus-one/66-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        
        if digits[-1] != 9:
            digits[-1] += 1
            
            return digits
        
        else:      
            sum = 0
            for d in digits:
                sum = sum*10 + d
            sum += 1

            res = []
            for s in str(sum):
                res.append(int(s))

            return res
        

File: 66-plus-one/66-plus-one.c
#include <stdio.h>
#include <stdlib.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* plusOne(int* digits, int digitsSize, int* returnSize) {

	*returnSize = digitsSize;

	if (digits[digitsSize-1] != 9) {
		digits[digitsSize - 1]++;
	}

	else {
		for (int i = digitsSize - 1; i >= 0; i--) {
			if (digits[i] == 9) 
				digits[i] = 0;
			else {
				digits[i]++;
				return digits;
			}
		}

		(*returnSize)++;
		digits = realloc(digits, (*returnSize) * sizeof(int));
		digits[*returnSize - 1] = 0;
		digits[0] = 1;
	}
    
    return digits;
    
}